apps/meteor/app/models/server/models/Subscriptions.js

```diff
@@ -12,8 +12,8 @@ export class Subscriptions extends Base {
 	constructor(...args) {
 		super(...args);
 
-		this.tryEnsureIndex({ rid: 1 });
 		this.tryEnsureIndex({ rid: 1, ls: 1 });
+		this.tryEnsureIndex({ 'rid': 1, 'userHighlights.0': 1 });
```

Comment:
I've found a query using this inside an `$or` at:

https://github.com/RocketChat/Rocket.Chat/blob/86136cea1fff6aa68ebb0c46208923bbac510856/apps/meteor/app/lib/server/lib/sendNotificationsOnMessage.js#L254

not sure if the new index will be used for that query as well 🤔